### PR TITLE
Commands: Try debouncing search for post-type navigation

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
PR tries to debounce search requests made by post-type navigation commands.

## Why?
Currently, the command is making a request for each keystroke, which can be expensive depending on the WP installation size.

## Testing Instructions
1. Open a post or page.
2. Open the Command palette.
3. Search for a post or a page by title.
4. Confirm that network requests are delayed.

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/240569/8b81ae52-f4e2-4939-ac7d-366b45d97aca

### CODE REVIEW

File: packages/core-commands/src/site-editor-navigation-commands.js [hunk 1]
```diff
@@ -33,17 +34,35 @@ const icons = {
 	wp_template_part: symbolFilled,
 };
 
+function useDebouncedValue( value ) {
+	const [ debouncedValue, setDebouncedValue ] = useState( '' );
+	const debounced = useDebounce( setDebouncedValue, 400 );
```
Comment (Mamaduka) [inline packages/core-commands/src/site-editor-navigation-commands.js]: Happy to adjust the delay.

**Comment by youknowriad** [comment]
Feels like we should use a similar delay on all of these useDebounceInput calls. Looks like the hardcoded value there is 250ms

**Comment by Mamaduka** [comment]
I am not sure; 400ms might be a bit long for the inserter. I guess it all depends on what we're debouncing.

**Comment by youknowriad** [comment]
I was thinking the opposite, update this PR to use 250?

**Comment by Mamaduka** [comment]
That makes more sense 😅

**Review by youknowriad — APPROVED** [review]

**Pull request merged**